src/kernel/coinstats.cpp

```diff
@@ -48,15 +48,35 @@ uint64_t GetBogoSize(const CScript& script_pub_key)
            script_pub_key.size() /* scriptPubKey */;
 }
 
-DataStream TxOutSer(const COutPoint& outpoint, const Coin& coin)
+template <typename T>
+static void TxOutSer(T& ss, const COutPoint& outpoint, const Coin& coin)
 {
-    DataStream ss{};
     ss << outpoint;
     ss << static_cast<uint32_t>(coin.nHeight * 2 + coin.fCoinBase);
```

Comment:
```suggestion
    ss << static_cast<uint32_t>((coin.nHeight << 1) + coin.fCoinBase);
```

This should avoid UBSan screaming at us in the future, e.g. `kernel/coinstats.cpp:55:46: runtime error: signed integer overflow: 2147483636 * 2 cannot be represented in type 'int'`.

Reply:
Hm, it isn't complaining. `nHeight` is also not signed.

Reply:
I have adopted @dergoegge 's suggestion for now, let me know if you would prefer a different change though @maflcko